flow/project.py

```diff
@@ -1174,7 +1176,87 @@ def highlight(s, eligible):
         context = self._get_standard_template_context()
 
         # get job status information
-        tmp = self._fetch_status(jobs, err, ignore_errors, no_parallelize)
+        if profile:
+            try:
+                import pprofile
+            except ImportError:
+                raise RuntimeWarning(
+                    "Profiling requires the pprofile package. "
+                    "Install with `pip install pprofile`.")
+            prof = pprofile.StatisticalProfile()
+
+            fn_filter = [
+                inspect.getfile(threading),
+                inspect.getfile(multiprocessing),
+                inspect.getfile(Pool),
+                inspect.getfile(ThreadPool),
+                inspect.getfile(tqdm),
+            ]
+
+            with prof(single=False):
+                tmp = self._fetch_status(jobs, err, ignore_errors, no_parallelize)
+
+            prof._mergeFileTiming()
+
+            # Unrestricted
+            total_impact = 0
+            total_num_hits = sum([hit[2] for fn, ft in prof.merged_file_dict.items()
```

Comment:
I think it would be cleaner to define hits first and then do something like
total_num_hits = `sum([hit[2] for hit in hits])`